apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx

```diff
@@ -0,0 +1,943 @@
+ *
+ * @param props - Props injected to the component.
+ * @returns Attribute Selection component.
+ */
+export const AttributeSelectionOIDC: FunctionComponent<AttributeSelectionOIDCPropsInterface> = (
+    props: AttributeSelectionOIDCPropsInterface
+): ReactElement => {
+
+    const {
+        claims,
+        externalClaims,
+        externalClaimsGroupedByScopes,
+        setExternalClaimsGroupedByScopes,
+        setExternalClaims,
+        selectedExternalClaims,
+        selectedDialect,
+        setSelectedExternalClaims,
+        selectedSubjectValue,
+        claimConfigurations,
+        defaultSubjectAttribute,
+        readOnly,
+        isUserAttributesLoading,
+        setUserAttributesLoading,
+        onlyOIDCConfigured,
+        [ "data-testid" ]: testId,
+        [ "data-testid" ]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const { getLink } = useDocumentation();
+
+    const [ availableExternalClaims, setAvailableExternalClaims ] = useState<ExtendedExternalClaimInterface[]>([]);
+
+    const [
+        filterSelectedExternalClaims,
+        setFilterSelectedExternalClaims
+    ] = useState<ExtendedExternalClaimInterface[]>([]);
+
+    const [
+        unfilteredExternalClaimsGroupedByScopes,
+        setUnfilteredExternalClaimsGroupedByScopes
+    ] = useState<OIDCScopesClaimsListInterface[]>(externalClaimsGroupedByScopes);
+
+    const [ expandedScopes, setExpandedScopes ] = useState<string[]>([]);
+
+    const [ searchValue, setSearchValue ] = useState<string>("");
+
+    const [ initializationFinished, setInitializationFinished ] = useState<boolean>(false);
+
+    const initValue = useRef(false);
+
+    const config: ConfigReducerStateInterface = useSelector((state: AppState) => state.config);
+
+    const [ openIDConnectClaims ] = useState<ExtendedExternalClaimInterface[]>(externalClaims);
+
+    useEffect(() => {
+        const tempFilterSelectedExternalClaims = [ ...filterSelectedExternalClaims ];
+
+        claimConfigurations?.claimMappings?.map((claim: ClaimMappingInterface) => {
+            if (
+                !filterSelectedExternalClaims.find(
+                    (selectedExternalClaim) => selectedExternalClaim?.mappedLocalClaimURI === claim.localClaim.uri
+                )
+            ) {
+                const availableExternalClaim = availableExternalClaims.find(
+                    (availableClaim) => availableClaim?.mappedLocalClaimURI === claim.localClaim.uri
+                );
+
+                if (availableExternalClaim) {
+                    tempFilterSelectedExternalClaims.push(availableExternalClaim);
+                }
+            }
+        });
+
+        setSelectedExternalClaims(tempFilterSelectedExternalClaims);
+        setFilterSelectedExternalClaims(tempFilterSelectedExternalClaims);
+    }, [ claimConfigurations ]);
+
+    // Stops the UI loader when the component is initialized and have fetched the availableExternalClaims.
```

Comment:
Make this a doc comment.

Reply:
resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611